src/cli/commands/outdated.js

```diff
@@ -22,12 +26,29 @@ export function hasWrapper(commander: Object, args: Array<string>): boolean {
 export async function run(config: Config, reporter: Reporter, flags: Object, args: Array<string>): Promise<number> {
   const lockfile = await Lockfile.fromDirectory(config.lockfileFolder);
   const install = new Install({...flags, includeWorkspaceDeps: true}, config, reporter, lockfile);
+  const {main, dev, optional} = flags;
   let deps = await PackageRequest.getOutdatedPackages(lockfile, install, config, reporter);
 
+  if ((main && dev) || (main && optional) || (dev && optional)) {
+    throw new MessageError(reporter.lang('yarnOutdatedMultipleFlags', config.cwd));
+  }
+
   if (args.length) {
     const requested = new Set(args);
 
     deps = deps.filter(({name}) => requested.has(name));
+
+    if (main || dev || optional) {
+      reporter.info(reporter.lang('yarnOutdatedFlagsWithPackage'));
```

Comment:
Maybe this isn't important to warn about? The specified package will be checked and that's cool. The flag is just a noop in that case.